remapping_pipeline.nf

```diff
@@ -39,6 +50,94 @@ newgenome_fai = file("${params.newgenome}.fai")
 outfile_basename = file(params.outfile).getName()
 outfile_dir = file(params.outfile).getParent()
 
+/*
+* Index the new reference genome using bowtie_build
+*/
+if ( !file("${newgenome_dir}/${newgenome_basename}.1.bt2").exists()){
+
+    process bowtieGenomeIndex {
+        // Memory required is 10 times the size of the fasta in Bytes or at least 1GB
+        memory Math.max(file(params.newgenome).size() * 10, 1073741824) + ' B'
+
+        publishDir newgenome_dir,
+            overwrite: false,
+            mode: "move"
+
+        input:
+            path "genome_fasta" from params.newgenome
+
+        output:
+            path "$newgenome_basename.*.bt2" 
+
+        """
+        bowtie2-build genome_fasta $newgenome_basename
+        """
+    }
+}
+
+/*
+* Check that the fai index file for old genome exists and if it does not create it.
+* Once created the publishDir directive will place it in the location described by the oldgenome_fai variable
+*/
+if (!oldgenome_fai.exists()){
+    process samtoolsFaidxOld {
+
+        publishDir oldgenome_dir,
+            overwrite: false,
+            mode: "copy"
+
+        input:
+            path "${oldgenome_basename}" from params.oldgenome
+
+        output:
+            path "${oldgenome_basename}.fai" into oldgenome_fai
+
+        """
+        samtools faidx ${oldgenome_basename}
+        """
+    }
+}
+
+/*
+* Check that the fai index file for new genome exists and if it does not create it.
+* Once created the publishDir directive will place it in the location described by the newgenome_fai variable
+*/
+if (!newgenome_fai.exists()){
+    process samtoolsFaidxNew {
+
+        publishDir newgenome_dir,
+            overwrite: false,
+            mode: "copy"
+
+        input:
+            path "${newgenome_basename}" from params.newgenome
+
+        output:
+            path "${newgenome_basename}.fai" into newgenome_fai
+
+        """
+        samtools faidx ${newgenome_basename}
+        """
+    }
+}
```

Comment:
Currently, samtools indexer processes are duplicated for old and new genome. Looks like this can be solved with [input repeaters](https://www.nextflow.io/docs/latest/process.html#input-repeaters).

If the problem with using input repeaters is conditionals (i.e. we want to check whether the index already exists separately for both genomes), then this conditional can be moved inside the actual command to be executed in this form:

```bash
[ ! -f ${newgenome_basename}.fai && samtools faidx ${newgenome_basename} ]
```

Reply:
Additionally, I'd like to point out that relying on preexisting indexes might be dangerous — for example, if someone had their FASTA indexed with samtools version X, and we use samtools version Y in our pipeline, it might cause hard to debug conflicts. So maybe we would be better off always generating index from fresh, as part of the processing pipeline.

Reply:
Ii've not addressed this particular comment because having bot process in one causes some issue in the output channel. (specifically the order in the output channel needs to be known so that the right `.fai` can be used)
This might be due to y limited understanding of how nextflow works at this time but for now I would rather keep the explicit duplication.
About the second point. There is always a risk about using externally constructed index (with samtools/bowtie or others) but most tools have been pretty good at not changing their indexes. So the risk is limited and the saving time/resources can be big.

Reply:
I agree this is not critical and shouldn't be blocking merging of this PR. However, I still think the refactoring of duplicating code fragments and discussing whether it's always safe to use external indexes still warrant a separate look. I've created a follow-up issue: #11.